lib/modules/manager/cargo/types.ts

```diff
@@ -38,3 +38,31 @@ export interface CargoRegistries {
   // maps registry names to URLs
   [key: string]: string;
 }
+
+export interface LockFilePackage {
+  name: string,
+  version: string,
+  source?: string, // omitted in internal deps like `cargo`
+  checksum?: string, // not available in v1
+  dependencies?: string[], // omitted if none
+}
+
+export interface LockFileBase {
+  package?: LockFilePackage[]
+}
+
+// version 1 stores checksums in `metadata`
+export interface LockFile1 extends LockFileBase {
+  metadata?: Record<string, string>
+}
+
+// version 2 stores checksums in `packages`
+export interface LockFile2 extends LockFileBase {
+}
+
+// version 3 is identified by `version` and fixes default git branch handling
+export interface LockFile3 extends LockFileBase {
+  version: 3;
+}
+
+export type LockFile = LockFile3 | LockFile1 | LockFile2;
```

Comment:
It would probably be fine to merge all three versions as I don't think they don't differ in any way that matters for us.